sbom/examples/container_image/from_catalog.py

```diff
@@ -0,0 +1,222 @@
+                f"arch={image['architecture']}&repository_url={registry}&tag={tag}"
+            )
+            ref = {
+                "referenceCategory": "PACKAGE-MANAGER",
+                "referenceType": "purl",
+                "referenceLocator": purl,
+            }
+            image_pkg["externalRefs"].append(ref)
+
+        packages.append(image_pkg)
+        relationships.append(
+            {
+                "spdxElementId": "SPDXRef-image-index",
+                "relationshipType": "CONTAINS",
+                "relatedSpdxElement": spdx_image_id,
+            }
+        )
+
+        for rpm in get_rpms(catalog_image_id):
+            purl = (
+                f"pkg:rpm/redhat/{rpm['name']}@{rpm['version']}?"
+                # We don't have a way to find out which content set (RPM repo) an RPM came from,
+                # so we arbitrarily choose one here (assuming we have this mapping via RPM
+                # lockfiles or other means eventually).
+                f"arch={rpm['architecture']}&repository_url={content_sets[0]}"
+            )
+            srpm = rpm["srpm_nevra"].rstrip(".src")
+            spdx_rpm_id = f"SPDXRef-{rpm['architecture']}-{rpm['name']}"
+            rpm_pkg = {
+                "SPDXID": spdx_rpm_id,
+                "name": rpm["name"],
+                "versionInfo": rpm["version"],
+                "supplier": "Organization: Red Hat",
+                "downloadLocation": "NOASSERTION",
+                "packageFileName": rpm["nvra"] + ".rpm",
+                "licenseConcluded": "NOASSERTION",
+                "externalRefs": [
+                    {
+                        "referenceCategory": "PACKAGE-MANAGER",
+                        "referenceType": "purl",
+                        "referenceLocator": purl,
+                    },
+                    {
+                        "referenceCategory": "OTHER",
+                        "referenceType": "sbom_ref",
+                        # Or wherever else we host per-RPM SBOMs.
+                        "referenceLocator": (
+                            f"https://access.redhat.com/security/data/sbom/v1/rpm/"
+                            f"{srpm}.spdx.json.bz2"
+                        ),
+                    },
+                ],
+                # We don't have data on a checksum for binary RPMs included in images; should we?
+            }
+            packages.append(rpm_pkg)
+            relationships.append(
+                {
+                    "spdxElementId": spdx_image_id,
+                    "relationshipType": "CONTAINS",
+                    "relatedSpdxElement": spdx_rpm_id,
+                }
+            )
+
+    spdx = {
+        "spdxVersion": "SPDX-2.3",
+        "dataLicense": "CC0-1.0",
+        "SPDXID": "SPDXRef-DOCUMENT",
+        "creationInfo": {
+            "created": "2006-08-14T02:34:56-06:00",
+            "creators": [
+                "example SPDX document only",
```

Comment:
```suggestion
                "Tool: example SPDX document only",
```

minor fix for strict checkers like TPA